### PR TITLE
refactor(opentrons-ai-client): refactoring api call and inputprompt component

### PR DESCRIPTION
**Pull request by koji**
# Overview
refactoring api call and input prompt component
add a new package for testing axios

close AUTH-402

# Test Plan
1. `make teardown-js && make setup-js`
2. `make -C opentrons-ai-client dev`
3. open localhost:5173 (the port must be 5173 since the backend allows us to use only 5173 port)
4. log in (if you don't have an account, you will need to sign up)
5. send a prompt
check the app displays your prompt and gets the return from OpenAI API

# Changelog
- add custom hook for callApi and get access token
- export a function as a util function
- update paragraph text style to align the font styles

# Review requests

# Risk assessment
low

### CODE REVIEW

**Comment by koji** [comment]
- [x] add a custom hook for data fetch
- [x] add a custom hook for token retrieving
- [x] update tests
- [x] add test for `useGetAccessToken`
- [x] fix fetch functionality
- [x] merge edge

File: opentrons-ai-client/src/resources/utils/utils.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/**
+ * Calculates the number of lines in a given string.
+ * @param input - The string to calculate the number of lines for.
+ * @returns The number of lines in the input string.
+ */
+
+export const calcTextAreaHeight = (input: string): number => {
+  const rowsNum = input.split('\n').length
+  return rowsNum
```
Comment (ncdiehl11) [inline opentrons-ai-client/src/resources/utils/utils.ts]: nit, maybe just return the expression instead of assigning a const?

**Review by ncdiehl11 — COMMENTED** [review]
Code all looks good, just having trouble testing the response after posting a prompt. Maybe we can chat IRL about how to test?

File: opentrons-ai-client/src/resources/utils/utils.ts (hunk 1 above)
Comment (koji) [reply opentrons-ai-client/src/resources/utils/utils.ts]: good catch, thanks!

**Comment by Elyorcv** [comment]
@koji please rebase

**Review by y3rsh — APPROVED** [review]
Works for me against #15287

**Pull request merged**